test/map/map.test.cpp

```diff
@@ -1389,4 +1389,48 @@ TEST(Map, KeepRenderData) {
         test.frontend.render(test.map);
         EXPECT_EQ(resourcesCount * i, requestsCount);
     }
-}
\ No newline at end of file
+}
+
+namespace {
+
+bool isInsideTile(const mapbox::geometry::box<float>& box, float padding, Size viewportSize) {
+    if (box.min.x - padding < 0) return false;
+    if (box.min.y - padding < 0) return false;
+    if (box.max.x - padding > viewportSize.width) return false;
+    if (box.max.y - padding > viewportSize.height) return false;
+    return true;
+}
+
+} // namespace
+
+TEST(Map, PlacedSymbolData) {
+    MapTest<> test{std::move(MapOptions().withMapMode(MapMode::Tile))};
+
+    test.fileSource->tileResponse = makeResponse("vector.tile", true);
+    test.fileSource->glyphsResponse = makeResponse("glyphs.pbf", true);
+    test.fileSource->styleResponse = makeResponse("style_vector.json");
+    test.fileSource->sourceResponse = makeResponse("source_vector.json");
+    test.fileSource->spriteJSONResponse = makeResponse("sprite.json");
+    test.fileSource->spriteImageResponse = makeResponse("sprite.png");
+
+    // Camera options will give exactly one tile (12/1171/1566)
+    test.map.jumpTo(CameraOptions().withZoom(12).withCenter(LatLng{38.917982, -77.037603}));
+    test.map.getStyle().loadURL("mapbox://streets");
+    Size viewportSize = test.frontend.getSize();
+    test.frontend.getRenderer()->collectPlacedSymbolData(true);
+    test.frontend.render(test.map);
+
+    const auto& placedSymbols = test.frontend.getRenderer()->getPlacedSymbolsData();
+    EXPECT_FALSE(placedSymbols.empty());
+
+    bool foundPlacedInsideTile = false;
+    for (const auto& placedSymbol : placedSymbols) {
+        if (placedSymbol.textPlaced && placedSymbol.textCollisionBox) {
+            if (isInsideTile(*placedSymbol.textCollisionBox, placedSymbol.viewportPadding, viewportSize)) {
+                EXPECT_FALSE(placedSymbol.intersectsTileBorder);
+                foundPlacedInsideTile = true;
```

Comment:
This could be a counter no? Also I would count the `!isInsdeTile`. Both should be deterministic here.

Reply:
Also I would add a test for `collectPlacedSymbolData(false)` not generating any data.